src/main/java/com/direwolf20/mininggadgets/client/particles/laserparticle/LaserParticle.java

```diff
@@ -80,10 +81,14 @@ public LaserParticle(ClientLevel world, double d, double d1, double d2, double x
         xo = x;
         yo = y;
         zo = z;
-        RenderBlockTileEntity te = (RenderBlockTileEntity) world.getBlockEntity(new BlockPos((int)this.x, (int)this.y, (int)this.z));
+        BlockEntity te = world.getBlockEntity(new BlockPos((int)this.x, (int)this.y, (int)this.z));
         if (te != null) {
```

Comment:
Would it be simpler to do an `instanceof RenderBlockTileEntity tile` here, and an else check for the default case? Then the check could handle both scenarios nicely.